conans/client/generators/cmake_common.py

```diff
@@ -775,26 +843,7 @@ def _conan_basic_setup_common(addtional_macros, cmake_multi=False):
     include(${CMAKE_CURRENT_LIST_DIR}/conanbuildinfo_relwithdebinfo.cmake)
 endif()
 
-macro(conan_set_vs_runtime)
-    # This conan_set_vs_runtime is MORE opinionated than the regular one. It will
-    # Leave the defaults MD (MDd) or replace them with MT (MTd) but taking into account the
-    # debug, forcing MXd for debug builds. It will generate MSVCRT warnings if the dependencies
-    # are installed with "conan install" and the wrong build time.
-    if(CONAN_LINK_RUNTIME MATCHES "MT")
```

Comment:
Note this implementation is NOT the same as the other, so might not be factorized to a common one.

Reply:
I was looking for a test that justifies these two implementations, but none has failed (using `#tags: slow`). If this is intended, maybe we should add a test to show why we want it to be this way (I would vote for the _opinionated_ version for our helpers).

For this PR I will keep it without changes, I'll ask you to review it once it is ready.

On the other hand we are mentioning in the docs the variable `CONAN_LINK_RUNTIME_MULTI` ([here](https://docs.conan.io/en/latest/creating_packages/package_approaches.html?highlight=conan_link_runtime#n-configs-1-package) in an **Important note**), but I cannot find that variable in the sources... we probably need to dedicate some time to it (docs and/or sources), I feel like it has something to do with this `cmake_macros_multi` variable.

Reply:
TBD here: https://github.com/conan-io/conan/issues/6238